File: app/src/main/java/com/levnovikov/postbus/root/home/prebooking/poi_selector/PoiSelectorRouter.java
package com.levnovikov.postbus.root.home.prebooking.poi_selector;

import android.os.Parcelable;

import com.levnovikov.postbus.root.home.prebooking.poi_selector.di.PoiSelectorScope;
import com.levnovikov.system_base.state.NodeState;
import com.levnovikov.system_base.Router;

import javax.inject.Inject;

import io.reactivex.annotations.Nullable;

/**
 * Created by lev.novikov
 * Date: 20/12/17.
 */

@PoiSelectorScope
public class PoiSelectorRouter extends Router {

    @Inject
    public PoiSelectorRouter() {

    }

    @Override
    protected void destroyNode() {

    }

    @Override
    public NodeState getNodeState(@Nullable Parcelable stateData) {
        return NodeState.create(this.getClass(), null);
    }

    @Override
    public void setState(NodeState state) {

    }
}

File: app/src/main/java/com/levnovikov/postbus/root/home/prebooking/poi_selector/PoiSelectorRouter.kt
package com.levnovikov.postbus.root.home.prebooking.poi_selector

import com.levnovikov.postbus.root.home.prebooking.poi_selector.di.PoiSelectorScope
import com.levnovikov.system_base.NodeHolder
import com.levnovikov.system_base.Router
import com.levnovikov.system_base.node_state.NodeState

import javax.inject.Inject

/**
 * Created by lev.novikov
 * Date: 20/12/17.
 */

@PoiSelectorScope
class PoiSelectorRouter @Inject
constructor() : Router() {

    override val holders: Set<NodeHolder<*>> = setOf()

    override fun setState(state: NodeState) {

    }
}
